From 15618b3810ff5d2c24da242b689add554bf9164d Mon Sep 17 00:00:00 2001
From: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Date: Wed, 19 Mar 2025 14:21:40 +0100
Subject: [PATCH] counter: stm32-lptimer-cnt: drop pm runtime ops as handled in
 mfd core

Now that the stm32-lptimer MFD driver manages the clock with runtime PM,
it can be removed from here. It's now redundant.

Signed-off-by: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Change-Id: I375a3ef25b5f05814508e0bf2cf785cbec56d4fd
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/445402
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Olivier MOYSAN <olivier.moysan@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/counter/stm32-lptimer-cnt.c | 22 ----------------------
 1 file changed, 22 deletions(-)

--- a/drivers/counter/stm32-lptimer-cnt.c
+++ b/drivers/counter/stm32-lptimer-cnt.c
@@ -541,30 +541,8 @@ static int stm32_lptim_cnt_resume(struct
 	return 0;
 }
 
-static int stm32_lptim_cnt_runtime_suspend(struct device *dev)
-{
-	struct stm32_lptim_cnt *priv = dev_get_drvdata(dev);
-
-	clk_disable(priv->clk);
-
-	return 0;
-}
-
-static int stm32_lptim_cnt_runtime_resume(struct device *dev)
-{
-	struct stm32_lptim_cnt *priv = dev_get_drvdata(dev);
-	int ret;
-
-	ret = clk_enable(priv->clk);
-	if (ret)
-		dev_err(dev, "failed to enable clock. Error [%d]\n", ret);
-
-	return ret;
-}
-
 static const struct dev_pm_ops stm32_lptim_cnt_pm_ops = {
 	SYSTEM_SLEEP_PM_OPS(stm32_lptim_cnt_suspend, stm32_lptim_cnt_resume)
-	RUNTIME_PM_OPS(stm32_lptim_cnt_runtime_suspend, stm32_lptim_cnt_runtime_resume, NULL)
 };
 
 static const struct of_device_id stm32_lptim_cnt_of_match[] = {
